src/main/java/io/rebloom/client/cf/Cuckoo.java

```diff
@@ -0,0 +1,193 @@
+   * options
+   *
+   * @param key     The name of the filter
+   * @param options An instance of CFInsertOptions containing the options
+   *                (CAPACITY/NOCREATE)
+   * @param items    One or more items to add
+   * @return true if the item was successfully inserted, false if an error
+   *         occurred
+   */
+  List<Boolean> cfInsert(String key, CFInsertOptions options, String... items);
+
+  /**
+   * CF.INSERTNX Adds one or more items to a cuckoo filter, only if it does not
+   * exist yet
+   *
+   * @param key   The name of the filter
+   * @param items One or more items to add
+   * @return true if the item was added to the filter, false if the item already
+   *         exists.
+   */
+  List<Boolean> cfInsertNx(String key, String... items);
+
+  /**
+   * CF.INSERTNX Adds one or more items to a cuckoo filter, using the passed
+   * options
+   *
+   * @param key     The name of the filter
+   * @param options An instance of CFInsertOptions containing the options
+   *                (CAPACITY/NOCREATE)
+   * @param items   One or more items to add
+   * @return true if the item was added to the filter, false if the item already
+   *         exists.
+   */
+  List<Boolean> cfInsertNx(String key, CFInsertOptions options, String... items);
+
+  /**
+   * CF.EXISTS Check if an item exists in a Cuckoo Filter
+   *
+   * @param key  The name of the filter
+   * @param item The item to check for
+   * @return false if the item certainly does not exist, true if the item may
+   *         exist.
+   */
+  boolean cfExists(String key, String item);
+
+  /**
+   * CF.DEL Deletes an item once from the filter. If the item exists only once, it
+   * will be removed from the filter. If the item was added multiple times, it
+   * will still be present.
+   *
+   * @param key  The name of the filter
+   * @param item The item to delete from the filter
+   * @return true if the item has been deleted, false if the item was not found.
+   */
+  boolean cfDel(String key, String item);
+
+  /**
+   * CF.COUNT Returns the number of times an item may be in the filter.
+   *
+   * @param key  The name of the filter
+   * @param item The item to count
+   * @return The number of times the item exists in the filter
+   */
+  long cfCount(String key, String item);
+
+  /**
+   * CF.SCANDUMP Begins an incremental save of the cuckoo filter. This is useful
+   * for large cuckoo filters which cannot fit into the normal SAVE and RESTORE
+   * model.
+   *
+   * The Iterator is passed as input to the next invocation of SCANDUMP . If
+   * Iterator is 0, the iteration has completed.
+   *
+   * @param key      Name of the filter
+   * @param iterator This is either 0, or the iterator from a previous invocation
+   *                 of this command
+   * @return a Map.Entry containing the Iterator and Data.
+   */
+  Map.Entry<Long, byte[]> cfScanDump(String key, Long iterator);
```

Comment:
*"This is either 0, or the iterator from a previous invocation"* - seems it doesn't have to be `Long`.
```suggestion
  Map.Entry<Long, byte[]> cfScanDump(String key, long iterator);
```